### ISSUE TITLE
NPE in ResponseListeners content notification

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#10726]
**Jetty version(s)**
12+

**Description**
Reported in https://github.com/jetty-project/jetty-reactive-httpclient/issues/296.

This probably happens with `Transfer-Encoding: chunked`, when there is data to read, but it is only the chunk metadata, as in:

```
HTTP/1.1 200 OK
Transfer-Encoding: chunked

10
```

In this case, there will be a wakeup of the demand callback because there is data, but calling `response.read()` would yield `null`.

This case is not covered in the case of multiple response listeners in `ResponseListeners`.

```
[HttpClient@1623b78d-37] WARN org.eclipse.jetty.util.thread.SerializedInvoker - Serialized invocation error
java.lang.NullPointerException: Cannot invoke "org.eclipse.jetty.io.Content$Chunk.hasRemaining()" because "chunk" is null
	at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer$ContentSource.onChunk(ResponseListeners.java:605)
	at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer.onDemandCallback(ResponseListeners.java:492)
	at org.eclipse.jetty.util.thread.SerializedInvoker$Link.run(SerializedInvoker.java:191)
	at org.eclipse.jetty.util.thread.SerializedInvoker.run(SerializedInvoker.java:117)
	at org.eclipse.jetty.client.transport.HttpReceiver$ContentSource.invokeDemandCallback(HttpReceiver.java:722)
	at org.eclipse.jetty.client.transport.HttpReceiver$ContentSource.onDataAvailable(HttpReceiver.java:661)
	at org.eclipse.jetty.client.transport.HttpReceiver.responseContentAvailable(HttpReceiver.java:303)
	at org.eclipse.jetty.client.transport.internal.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:82)
	at org.eclipse.jetty.client.transport.internal.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:90)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:194)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.ssl.SslConnection$SslEndPoint.onFillable(SslConnection.java:554)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:373)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:146)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:410)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

**Comment by olenagerasimova** [comment]
Hi again, the problem does not seem completely fixed, jetty **12.0.3** fails with
```
java.lang.IllegalStateException
at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer$ContentSource.demand(ResponseListeners.java:665)
at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.read(ResponseListenerProcessor.java:197)
at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.run(ResponseListenerProcessor.java:181)
at org.eclipse.jetty.reactive.client.internal.QueuedSinglePublisher.tryProduce(QueuedSinglePublisher.java:70)
at org.eclipse.jetty.reactive.client.internal.ResponseListenerProcessor$ContentPublisher.onRequest(ResponseListenerProcessor.java:174)
at org.eclipse.jetty.reactive.client.internal.AbstractSinglePublisher.request(AbstractSinglePublisher.java:90)
```

I've got java 17 application running inside docker container and performing some requests to maven-central. If this same application is running locally (not inside container), no problems occur.

**Comment by lorban** [comment]
@olenagerasimova This seems to be caused by a different problem. I tried running the reproducer you included in https://github.com/jetty-project/jetty-reactive-httpclient/issues/296 and it passes with `jetty-reactive-httpclient:4.0.1`.

You should open a new [jetty-reactive-httpclient](https://github.com/jetty-project/jetty-reactive-httpclient) issue with a new reproducer if possible.

Thanks!
